0001-kaa-protocol/README.md

```diff
@@ -93,40 +106,88 @@ Examples of extension specific paths are:
 /json/status
 ```
 
-### Endpoint-specific and nonspecific extensions
+### Payload format specifier
+Extensions may support multiple payload formats.
+In that case, it is wise to add payload format specifier.
 
-There are two large classes of extensions: those that are endpoint-specific and require an endpoint to operate on, and those that do not. While most extensions are endpoint-specific (data collection, profiling, configuration, etc.), there are extensions that are not (endpoint registration).
+For example, use `/json` for JSON-formatted payload, and `/protobuf/<scheme_id>` for protocol buffers.
 
-Endpoint-specific extensions should start path with an endpoint token. Nonspecific extension should not.
+// TODO: CoAP has Content-Format option for that. It already has json and cbor, but not protobuf.
 
-### Payload format specifier
+### Request/response pattern
+Many extensions require request/response style communication, which is not supported by MQTT natively.
+That is usually overcome by introducing a separate topic name for responses.
 
-Extensions may support multiple payload formats. In that case, it is wise to add payload format specifier.
+We introduce `/status` topic for response messages.
+In case original payload contains `packet_id`, response should be published in that topic.
 
-For example, use `/json` for JSON-formatted payload, and `/protobuf/<scheme_id>` for protocol buffers.
+**Note:** this only applies to client-originating requests.
```

Comment:
Why server uses different request/response schema?

Can we make the /status is always the response (except for specific cases where response is something more complex) for both client- and server- side originating requests?